privatelink/gcp/terraform/privatelink.tf

```diff
@@ -0,0 +1,140 @@
+terraform {
+  required_version = ">= 0.12.17"
+  required_providers {
+    google = {
+      source  = "hashicorp/google"
+      version = "4.11.0"
+    }
+    external = {
+      source  = "hashicorp/external"
+      version = "2.0.0"
+    }
+    time = {
+      source  = "hashicorp/time"
+      version = "0.6.0"
+    }
+  }
+}
+
+provider "google" {
+  project = var.project
+  region  = var.region
+}
+
+variable "project" {
+  description = "The GCP project to provision"
+  type        = string
+}
+
+variable "region" {
+  description = "The Region to provision"
+  type        = string
+}
+
+variable "network_name" {
+  description = "The Network Name to provision Private Service Connect endpoint to Confluent Cloud"
+  type        = string
+}
+
+variable "subnetwork_name" {
+  description = "The Subnetwork Name to provision Private Service Connect endpoint to Confluent Cloud"
+  type        = string
+}
+
+variable "bootstrap" {
+  description = "The bootstrap server (ie: lkc-abcde-vwxyz.us-central1.gcp.glb.confluent.cloud:9092)"
+  type        = string
+}
+
+variable "psc_service_attachments_by_zone" {
+  description = "A map of Zone to Service Attachment from Confluent Cloud to Private Service Connect with (provided by Confluent)"
+  type        = map(string)
+}
+
+locals {
+  hosted_zone = replace(regex("^[^.]+-([0-9a-zA-Z]+[.].*):[0-9]+$", var.bootstrap)[0], "glb.", "")
+  network_id  = regex("^([^.]+)[.].*", local.hosted_zone)[0]
```

Comment:
qq: is it a fancy way to infer confluent cloud network's ID that a cluster belongs to? For example, `network_id` would be `n-gjd4og` here:
![image](https://user-images.githubusercontent.com/5459870/178612902-f98fc591-ae4a-4769-b9fa-1753b9775e36.png)